src/video.c

```diff
@@ -180,6 +185,17 @@ struct vidqent {
 };
 
 
+struct vidqdec {
+	struct le le;
+	struct rtp_header *hdr;
+	struct mbuf *mb;
+	struct vrx vrxc;
```

Comment:
Most of the vrx fields are not used in the `vrx_thread`. Should we separate those in another struct?
```c
vrxc->n_intra
vrxc->size
vrxc->fmt
vrxc->filtl  /* This maybe better taken from vrxo. */
vrxc->stats.disp_frames
vrxc->frames
vrxc->ts_recv
```
But what do think about the following proposal?
- Start an extra re polling loop for the rtp_listen socket in the `vrx_thread`.
- Thus RTP read, decoding, filter and display all is done in the `vrx_thread`.
- The `vrx_thread` "owns" the vrx object.
- Changes of `vrx` from main thread have to be reported by means of a `vrx_update` object. Only `vrx_update` has to be locked. It is copied before it is processed in the `vrx_thread`.
- The data reported from `vrx_thread` to main thread is reported with a `vrx_info` object to the main thread. Only `vrx_info` has to be locked. It is copied before it is processed in the main thread.

Reply:
> Most of the vrx fields are not used in the `vrx_thread`. Should we separate those in another struct?

Maybe within separate PR. 

> But what do think about the following proposal?
> Start an extra re polling loop for the rtp_listen socket in the vrx_thread

For this PR I would like to keep the scope small. But in general the most problematic part are the udp helpers (DTLS/SRTP etc.) I think. These helpers have `tmr` objects and need to be transferred to the dedicated `vrx_thread` re loop then. Or everything has to be setuped from `vrx_thread` directly. Maybe not the only part that needs to be handled for full thread safety here. Another problem is maybe BUNDLE (audio/video and RTP/RTCP share one RTP connection).

Reply:
Ah, I see. I keep this in mind when starting with RX audio thread.

So let's then proceed with this video PR. The `vrxc` and `vrxo` helps to reduce long blocking locks now, right? The `lock_rx` is locked only short in both threads. So no decoding, display is done while the lock is hold.

While the `lock_dec` is hopefully not very often locked by the main thread.
-  Switching off the video RX might block while one (last) video frame is decoded and sent to the display.

More concretely:
- video_stop_display(). Will be called if the user switches off video. This could lead to a long blocking situation and cause an audio artifact.
- video_decoder_set(). Very seldom, only if codec changes.
- set_vidisp() only called by `video_start_display()`. Maybe will not be blocked by the decoding thread.

Maybe we could improve this in a follow up PR.

> Check filter (vidfilt) and display (vidisp) modules are thread safe too

Also with the `lock_dec` mutex? Updates of the filter list, decoder and display could be handled similar.